### DISCUSSION TITLE
ScriptReference/Microphone.Start

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#300] [Notes]
# ScriptReference/Microphone.Start

https://docs.unity3d.com/ScriptReference/Microphone.Start.html

**Comment by vertxxyz** [comment]
A robust way of recording, and restarting recording after a microphone disconnects and reconnects is to:
1. Wait a frame before calling `Start`.
1. Wait until `GetPosition` is `> 0` before playing your audio source.

If you don't wait a frame, you may get stuck recording nothing, where `Start` returns a clip, but `GetPosition` never moves, and the clip remains empty.
